bundles/org.openhab.binding.caddx/src/main/java/org/openhab/binding/caddx/internal/handler/LogEventMessage.java

```diff
@@ -0,0 +1,100 @@
+ */
+package org.openhab.binding.caddx.internal.handler;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.binding.caddx.internal.CaddxMessage;
+
+/**
+ * Used to parse panel log event messages.
+ *
+ * @author Georgios Moutsos - Initial contribution
+ */
+@NonNullByDefault
+public class LogEventMessage {
+
+    public final String number;
+    public final String size;
+    public final String type;
+    public final String zud;
+    public final String partition;
+    public final String month;
+    public final String day;
+    public final String hour;
+    public final String minute;
+
+    LogEventMessage(CaddxMessage message) {
+        this.number = message.getPropertyById("panel_log_event_number");
+        this.size = message.getPropertyById("panel_log_event_size");
+        this.type = message.getPropertyById("panel_log_event_type");
+        this.zud = message.getPropertyById("panel_log_event_zud");
+        this.partition = message.getPropertyById("panel_log_event_partition");
+        this.month = message.getPropertyById("panel_log_event_month");
+        this.day = message.getPropertyById("panel_log_event_day");
+        this.hour = message.getPropertyById("panel_log_event_hour");
+        this.minute = message.getPropertyById("panel_log_event_minute");
+    }
+
+    private final String alarmList[] = { "Alarm", "Alarm restore", "Bypass", "Bypass restore", "Tamper",
+            "Tamper restore", "Trouble", "Trouble restore", "TX low battery", "TX low battery restore", "Zone lost",
+            "Zone lost restore", "Start of cross time", "Not used", "Not used", "Not used", "Not used",
+            "Special expansion event", "Duress", "Manual fire", "Auxiliary 2 panic", "Not used", "Panic",
+            "Keypad tamper", "Control box tamper", "Control box tamper restore", "AC fail", "AC fail restore",
+            "Low battery", "Low battery restore", "Over-current", "Over-current restore", "Siren tamper",
+            "Siren tamper restore", "Telephone fault", "Telephone fault restore", "Expander trouble",
+            "Expander trouble restore", "Fail to communicate", "Log full", "Opening", "Closing", "Exit error",
+            "Recent closing", "Auto-test None", "Start program", "End program", "Start download", "End download",
+            "Cancel", "Ground fault", "Ground fault restore", "Manual test", "Closed with zones bypassed",
+            "Start of listen in", "Technician on site", "Technician left", "Control power up", "58-", "", "", "", "",
+            "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "",
+            "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "",
+            "", "", "Not used", "First to open", "Last to close", "PIN entered with bit 7 set", "Begin walk-test",
+            "End walk-test", "Re-exit", "Output trip", "Data lost" };
```

Comment:
There needs to be a way to understand at a glance what message corresponds to which event type, otherwise it is extremely difficult to make changes. I don't really know the best way to approach this but I'll leave it up to you to decide.
Here are a couple of ways you could do it:
* Populate the list in a method by doing `alarmList[1] = "Alarm restore"` for each index
* Put the alarm list into a properties file and use event type as the keys. If you wanted this could also open up the possibility of having different property files for different languages.

Reply:
The class has been refactored to use a new LogEventType enum, which packs the knowledge of the specific log events.